Mathlib/Topology/Algebra/InfiniteSum/Defs.lean

```diff
@@ -59,121 +67,237 @@ section HasProd
+def HasProd (f : β → α) (a : α) (L := unconditional β) : Prop :=
+  Tendsto (fun s : Finset β ↦ ∏ b ∈ s, f b) L.filter (𝓝 a)
+
+/-- `Multipliable f` means that `f` has some (infinite) product with respect to `L`. Use `tprod` to
+get the value. -/
 @[to_additive
-/-- `Summable f` means that `f` has some (infinite) sum. Use `tsum` to get the value. -/]
-def Multipliable (f : β → α) : Prop :=
-  ∃ a, HasProd f a
+/-- `Summable f` means that `f` has some (infinite) sum with respect to `L`. Use `tsum` to get the
+value. -/]
+def Multipliable (f : β → α) (L := unconditional β) : Prop :=
+  ∃ a, HasProd f a L
+
+@[to_additive]
+lemma Multipliable.mono_filter {f : β → α} {L₁ L₂ : SummationFilter β}
+    (hf : Multipliable f L₂) (h : L₁.filter ≤ L₂.filter) : Multipliable f L₁ :=
+  match hf with | ⟨a, ha⟩ => ⟨a, ha.mono_left h⟩
 
 open scoped Classical in
-/-- `∏' i, f i` is the product of `f` if it exists and is unconditionally convergent,
-or 1 otherwise. -/
-@[to_additive /-- `∑' i, f i` is the sum of `f` if it exists and is unconditionally convergent,
-or 0 otherwise. -/]
-noncomputable irreducible_def tprod {β} (f : β → α) :=
-  if h : Multipliable f then
-  /- Note that the product might not be uniquely defined if the topology is not separated.
-  When the multiplicative support of `f` is finite, we make the most reasonable choice to use the
-  product over the multiplicative support. Otherwise, we choose arbitrarily an `a` satisfying
-  `HasProd f a`. -/
-    if (mulSupport f).Finite then finprod f
+/-- `∏' i, f i` is the unconditional product of `f`, if it exists, or 1 otherwise. ]
+
+More generally, if `L` is a `SummationFilter`, `∏'[L] i, f i` is the product of `f` with respect to
+`L` if it exists, and `1` otherwise.
+
+(Note that even if the unconditional product exists, it might not be unique if the topology is not
+separated. When the multiplicative support of `f` is finite, we make the most reasonable choice,
+to use the product over the multiplicative support. Otherwise, we choose arbitrarily an `a`
+satisfying `HasProd f a`. Similar remarks apply to more general summation filters.) -/
+@[to_additive /-- `∑' i, f i` is the unconditional sum of `f` if it exists, or 0 otherwise.
+
+More generally, if `L` is a `SummationFilter`, `∑'[L] i, f i` is the sum of `f` with respect to
+`L` if it exists, and `1` otherwise.
+
+(Note that even if the unconditional sum exists, it might not be unique if the topology is not
+separated. When the support of `f` is finite, we make the most reasonable choice, to use the product
+over the support. Otherwise, we choose arbitrarily an `a` satisfying `HasProd f a`. Similar remarks
+apply to more general summation filters.)
+-/]
+noncomputable irreducible_def tprod (f : β → α) (L := unconditional β) :=
+  if h : Multipliable f L then
+    if L.HasSupport ∧ (mulSupport f ∩ L.support).Finite then finprod (L.support.mulIndicator f)
     else h.choose
```

Comment:
Could you add here
```
  else if HasProd f 1 L then 1
  else h.choose
```
?

This is something we have seen with derivatives (but it took some time to converge to this): when there are several possible choices, if one can use 1 then it's often better than an arbitrary choice. For instance, it would make it possible to remove a bunch of `L.NeBot` assumptions in many statements.